### PR TITLE
FEATURE: Add Poetry integration

### PR DESCRIPTION
**Pull request by ucokzeko**
- [x] `Poetry` integration for generated project
- [x] Integrate `Poetry` to Surround CLI commands
- [x] Add a basic sample generated from the CLI tool that can be used for a template, including instructions to run the project without the CLI tool.

**Note**: If you run the sample project installing this version of the CLI tool, you'll need to install Poetry manually to your machine with `pip install poetry==1.1.14`

### CODE REVIEW

File: surround/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 pyaml==17.12.1
```
Comment (ScottyB) [inline surround/requirements.txt:1]: `pyaml` is installed twice when running `poetry install`. Do we need both versions?

File: examples/simple_example/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# simple_example
+
+## Introduction
+This example is generated from the Surround CLI without the web runner support.
```
Comment (ScottyB) [inline examples/simple_example/README.md:4]: What about adding instructions for how to use poetry with doit? Do we need to have doit and poetry?

Comment (ScottyB) [reply examples/simple_example/README.md:4]: Would be good if we can just use one tool.

File: examples/simple_example/main.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from simple_example.__main__ import main
+
```
Comment (ScottyB) [inline examples/simple_example/main.py:2]: What's this file for?

**Review by ScottyB — CHANGES_REQUESTED** [review]

File: surround/requirements.txt (hunk 1 above)
Comment (ucokzeko) [reply surround/requirements.txt:1]: Didn't realise that it's different from `pyyaml`. I wil have a look at this.

File: examples/simple_example/README.md (hunk 2 above)
Comment (ucokzeko) [reply examples/simple_example/README.md:4]: The only reason is that `doit` is only available in the CLI dependencies. If CLI is installed then we can just use the wrapped commands to run things.
But I can add instructions for `doit` by manually installing it if needed.

File: examples/simple_example/main.py (hunk 3 above)
Comment (ucokzeko) [reply examples/simple_example/main.py:2]: For the CI test to trigger the different examples

**Comment by ucokzeko** [comment]
Reenabled Tox and tests are working fine now

**Pull request merged**